### ISSUE TITLE
Disappearing chrs 

### ISSUE THREAD

**Issue by anazmy** [selectel/pyte#66]
Was doing a small test for an app I'm working on 

```
# -*- coding: utf-8 -*-
from __future__ import print_function, unicode_literals

from pyte import ByteStream,Screen,modes as mo


if __name__ == "__main__":
    screen = Screen(80, 24)
#    screen.reset_mode(mo.LNM)
    stream = ByteStream()
    stream.attach(screen)
    stream.feed(b"ipcs -s | grep nobody |awk '{print$2}'|xargs -n1 ipcrm sem ;ps aux|grep -P 'httpd|fcgi'|grep -v grep|awk '{print$2 \x0D}'|xargs kill -9;/etc/init.d/httpd startssl")

    for idx, line in enumerate(screen.display, 1):
        print("{0:2d} {1} ¶".format(idx, line))
```

And seems the CR (\x0D) is giving me wrong output  :

```
# python debug.py 
 1 ipcs -s | grep nobody |awk '{print$2}'|xargs -n1 ipcrm sem ;ps aux|grep -P 'http ¶
 2 }'|xargs kill -9;/etc/init.d/httpd startssl                                      ¶
 3                                                                                  ¶
 4                                                                                  ¶
```

Is that expected behavior or a bug ?

**Comment by anazmy** [comment]
that's Python 2.7.13  BTW

**Comment by superbobry** [comment]
Thanks for the bug report. Could you add which `pyte` version are you using?

**Comment by anazmy** [comment]
Sorry forgot to mention that, I initially used pyte-0.5.2 but even with latest `master` branch here I got same behavior .

Some context:

I'm capturing user input on xterm terminal (stdin) and sending it to pyte to handle the ctrl sequences , seems to me that when the input line is wider than the terminal - like above example- you hit this issue.

**Comment by anazmy** [comment]
I also tried reverting mode.LNM but got same behavior

**Comment by superbobry** [comment]
tl;dr this is correct behaviour.

Slightly longer version: `"\r"` is a carriage return symbol, which when encountered by the stream, cases the cursor to go to the beginning of line. Your input has a carriage return in the middle, therefore it is rendered as

```
ipcs -s | grep nobody |awk '{print$2}'|xargs -n1 ipcrm sem ;ps aux|grep -P 'http
```

then the screen goes to the new line and renders everything before `"\r"`

```
d|fcgi'|grep -v grep|awk '{print$2
                                 ^ 
                               cursor
```

on `"\r"` the cursor moves to the beginning of line

```
d|fcgi'|grep -v grep|awk '{print$2
^
cursor
```

and continues rendering remaining input from there

```
}'|xargs kill -9;/etc/init.d/httpd startssl
```

The resulting state is exactly what you're observing.

**Comment by anazmy** [comment]
Thanks for taking time to explain it helped, what do you think is the best course of action , considering that screen size and input length are variables ?

Should I remove `\r` from the byte stream before feeding pyte.stream ?

**Comment by anazmy** [comment]
Also shouldn't mode.LNM change behavior of the cursor in this case ?

**Comment by superbobry** [comment]
Roughly speaking `mode.LNM` (enabled by default) translates `"\n"` into `"\r\n"`, so it does not affect your case.

I think one possible workaround is to escape `"\r"` and related sequences, so they appear as two normal characters "\" and "r" and not a single control one.

**Comment by anazmy** [comment]
I will try your advice , thanks for the great work

**Comment by superbobry** [comment]
No problem, happy `pyte` is useful!
